scipy/datasets/_download_all.py

```diff
@@ -36,12 +37,20 @@ def download_all(path=None):
         raise ImportError("Missing optional dependency 'pooch' required "
                           "for scipy.datasets module. Please use pip or "
                           "conda to install 'pooch'.")
-    if path is None:
-        path = pooch.os_cache('scipy-data')
+
+    if path:
+        cache_dir = path
+    else:
+        cache_dir = pooch.os_cache('scipy-data')
```

Comment:
Should we also check for the permanent tmp file to read a path that might have been set before here?

 ```suggestion
    else:
        if os.path.exists(_registry._CACHE_DIR_FILE):
            with open(_registry._CACHE_DIR_FILE) as f:
                cache_dir = f.read().strip()
         else:
             cache_dir = pooch.os_cache('scipy-data')
```